controllers/update_status.go

```diff
@@ -607,33 +575,29 @@ func validateInstance(r *FoundationDBClusterReconciler, context ctx.Context, clu
 	}
 
 	processGroupStatus.UpdateCondition(fdbtypes.MissingPVC, incorrectPVC, cluster.Status.ProcessGroups, instanceID)
-	incorrectPod = incorrectPod || incorrectPVC
-
-	if incorrectPod {
-		return false, true, false, nil
-	}
 
 	var needsSidecarConfInConfigMap bool
 	for _, container := range instance.Pod.Spec.Containers {
 		if container.Name == "foundationdb" {
 			image := strings.Split(container.Image, ":")
 			version, err := fdbtypes.ParseFdbVersion(image[len(image)-1])
 			if err != nil {
-				return false, false, false, err
+				return false, err
 			}
 			if !version.PrefersCommandLineArgumentsInSidecar() {
 				needsSidecarConfInConfigMap = true
 			}
 		}
 	}
 
+	failing := false
 	for _, container := range instance.Pod.Status.ContainerStatuses {
-		if !container.Ready {
-			return true, false, needsSidecarConfInConfigMap, nil
-		}
+		failing = failing || !container.Ready
```

Comment:
If `failing` is true we can return early.
```suggestion
		if container.Ready {
		     continue
		}
		 
		 failing = true
		 break
```